src/plugins/grpc/client.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+
+const { FORMAT_HTTP_HEADERS, Tags } = require('opentracing')
+
+function patch (grpc, tracer, config) {
+  if (config.client === false) return
+
+  config = config.client || config
+
+  function gRPCInterceptor (options, nextCall) {
+    const { path, responseStream } = options.method_definition
+    const scope = tracer.scopeManager().active()
+    const span = tracer.startSpan('grpc.client', {
+      childOf: scope && scope.span(),
+      tags: {
+        [Tags.SPAN_KIND]: Tags.SPAN_KIND_RPC_CLIENT,
+        'grpc.method_name': path,
+        'grpc.method_type': responseStream ? 'server_stream' : 'unary',
+        'resource.name': path.split('/').slice(-1),
+        'service.name': config.service || `${tracer._service}-grpc`
+      }
+    })
+
+    return new grpc.InterceptingCall(nextCall(options), {
+      sendMessage: (message, next) => {
+        if (config.arguments) {
+          const values = config.arguments(message)
+
+          for (const key in values) {
+            span.setTag(`grpc.arguments.${key}`, values[key])
+          }
+        } else {
+          span.setTag(`grpc.arguments`, message)
+        }
+
+        next(message)
+      },
+
+      start: (metadata, _listener, next) => {
+        if (metadata) {
+          for (const [key, value] of Object.entries(metadata.getMap())) {
+            span.setTag(`grpc.metadata.${key}`, value)
+          }
+        }
+
+        // Inject tracing headers into the grpc call's metadata.
+        const tracingMetadata = metadata || new grpc.Metadata()
+        const meta = {}
+
+        tracer.inject(span, FORMAT_HTTP_HEADERS, meta)
+
+        for (const [key, value] of Object.entries(meta)) {
+          tracingMetadata.set(key, value)
+        }
+
+        next(tracingMetadata, {
+          onReceiveStatus: (status, next) => {
+            if (status.code !== 0) {
+              span.addTags({
+                'error.code': status.code,
+                'error.msg': status.details
+              })
+
+              if (config.arguments) {
+                const values = config.arguments(status.metadata.getMap())
+
+                for (const key in values) {
+                  span.setTag(`error.metadata.${key}`, values[key])
+                }
+              } else {
+                span.setTag(`error.metadata`, status.metadata.getMap())
+              }
+            }
+
+            span.finish()
+
+            next(status)
+          }
+        })
+      }
+    })
+  }
+
+  this.wrap(grpc.Client.prototype, 'resolveCallInterceptors', resolveCallInterceptors => {
+    return function () {
+      return [gRPCInterceptor, ...resolveCallInterceptors.call(this, arguments)]
+    }
+  })
+}
+
+function unpatch (grpc) {
+  this.unwrap(grpc.Client.prototype, 'resolveCallInterceptors')
+}
+
+module.exports = [{
+  name: 'grpc',
+  versions: ['>=1.13'],
+  patch,
+  unpatch
```

Comment:
It would be easier to read to clearly separate the patch from the wrapper factory, the wrapper and the wrapped function, similar to https://github.com/DataDog/dd-trace-js/blob/master/src/plugins/cassandra-driver.js

I know we have some integrations that don't do this 😅

Reply:
Similar to the above, feel free to ignore for now.